### ISSUE TITLE
Feature request: filter by HTTP method

### ISSUE THREAD

**Issue by zolakt** [mpalourdio/ng-http-loader#41]
It would be nice to be able to filter by HTTP methods
GET/HEAD/POST/PUT/DELETE

Basically, I'd like to use the loader globally for POST/PUT methods, but not for the most GET ones, since they are pretty fast and it just flashed for a fraction of a second.

**Comment by mpalourdio** [comment]
Could be possible. For fast requests, you can use the ``debounceDelay`` parameter, or exclude some url patterns.

Both are documented in the README.

**Comment by zolakt** [comment]
Tnx, I missed the `debounceDelay` option. I'll try with that.
But regardless, method filter could be useful.
Url patterns are ok, but are limited, especially in rest apis

**Comment by mpalourdio** [comment]
Could be useful eventually. Marking as feature, but I need to think about it.

**Comment by thienedits** [comment]
To add to this, sometimes on a page I have multiple rest apis I would like to run in the background to get new information. Sometimes these requests are on an Interval, so the spinner shows every time those requests are made. Not sure if this is even possible, but is it possible to send a flag per each http request to not show the spinner when this request is called?

**Comment by mpalourdio** [comment]
You can do this via https://github.com/mpalourdio/ng-http-loader#requests-filtering if you find a convenient pattern

**Comment by thienedits** [comment]
ah right, can just add some params to the request and filter using regex

**Comment by thienedits** [comment]
can you change the check in the intecept method to req.urlWithParams? I added query params and the check is currently not checking for params.

https://github.com/mpalourdio/ng-http-loader/blob/dd24009feb5e54aeb34f145f4c77b076953a8e0f/src/services/pending-interceptor.service.ts#L42

**Comment by mpalourdio** [comment]
@thienedits good catch, I will give it a shot ASAP

**Comment by mpalourdio** [comment]
@thienedits Waht kind of url with query params are you testing ? I have just tried ``req.url`` && ``req.urlwithParams``, and both output full url with query string for somenthing like ``/api/service?foo=bar``

I'm a bit confused here. Thanks for feedback

**Comment by mpalourdio** [comment]
Mhh, Ok, got it I think. You must use something like :

```typescript
this.http.post(
            'api/service1?test=toto',
            null,
            {
                'params': {
                    'foo': 'bar'
                }
            }
```

**Comment by thienedits** [comment]
@mpalourdio yes i add params like this 
```this.options.params = new HttpParams().set('hideLoader', 'true');
this.http.get(environment.apiUri + this.url + idString, this.options)
```

**Comment by mpalourdio** [comment]
@thienedits fixed in v0.7.1

**Comment by thienedits** [comment]
Awesome thanks!

**Comment by Troyd-Destin** [comment]
Came to support the idea of filtering by http method at a global level, using custom patterns is mildly annoying.

**Comment by mpalourdio** [comment]
Still on my todo list.

**Comment by mpalourdio** [comment]
WIP here : https://github.com/mpalourdio/ng-http-loader/tree/filter_methods

**Comment by mpalourdio** [comment]
Fixed in v2.2.0
